src/transformers/generation_utils.py

```diff
@@ -2009,21 +2023,24 @@ def contrastive_search(
             past_key_values = new_key_values
 
             logit_for_next_step = torch.stack(torch.split(logits, top_k))[range(bsz), selected_idx, :]
-            # contrastive_search main logic end::
-            # after running the above codes, we update following parameters: next_tokens, past_key_values,
-            # logit_for_next_step, selected_score, decoder_hidden_states_one_step
+
+            # Rebuilds the relevant parts of the model output for the selected token, for use in the next iteration
+            if self.config.is_encoder_decoder:
+                outputs = Seq2SeqLMOutput(
+                    past_key_values=past_key_values,
+                    decoder_hidden_states=decoder_hidden_states,
+                )
+            else:
+                outputs = CausalLMOutputWithPast(
+                    past_key_values=past_key_values,
+                    hidden_states=decoder_hidden_states,
+                    attentions=model_kwargs["attention_mask"],
+                )
```

Comment:
moves this above the "contrastive_search main logic end" comment, since it is unique to contrastive search